src/internal/components/TextInputInnerAction.tsx

```diff
@@ -93,7 +93,7 @@ const TextInputAction = forwardRef<HTMLButtonElement, TextInputActionProps>(
     return (
       <Box as="span" className="TextInput-action" marginLeft={1} marginRight={1} lineHeight="0">
         {icon && !children ? (
-          <Tooltip direction={tooltipDirection} aria-label={ariaLabel}>
+          <Tooltip direction={tooltipDirection} text={ariaLabel as unknown as string} type="label">
```

Comment:
`text={ariaLabel as unknown as string}` this is not great but some sort of type conversion or "hack" is needed here because `text` is a required prop for Tooltip but I see we can't make `ariaLabel` required for TextInput's trailing action because it could be an icon button or a text button. I am open to feedback if we want to handle this in another way

Reply:
Would this be a little easier to read/understand?

```jsx
text={ariaLabel || ''}
```

Reply:
Yeah that reads better. Thank you. Although both options will end up rendering an empty tooltip. I wonder if we should conditionally render the tooltip for icon button too? Though this solution will only work until we bring the tooltip to icon buttons by default https://github.com/primer/react/issues/2008. What are your thoughts on that?

Reply:
Ah ok. Yea, I think we can keep what you have then.